### PR TITLE
Fix Dependencies for Expo up to v46

### LINKED ISSUE #33: Projects depending on updated expo-linear-gradient conflict with package lock
**Describe the bug**

Our project depends on expo-linear-gradient@8.3.1, but this package-lock and published version of the package depends on ~8.3.0, causing a crash when both are loaded in react native.

**To Reproduce**
1. Make a new expo typescript project
2. Add expo-linear-gradient as a dependency
3. Add react-native-skeleton-content as a dependency
4. Make a View with react-native-skeleton-content and a gradient
5. expo start and see a crash

**Expected behavior**
No crash

**Screenshots**

**Smartphone (please complete the following information):**
 - iPhone X
 - iOS 14
 - 1.0.20

**Additional context**

### LINKED ISSUE #59: Dependencies versions upgrade for Expo SDK 44
Upgrading expo projects to expo SDK 44 added some changes in dependencies versions :
- expo-linear-gradient : 10.0.3 => 11.0.0
- react-native-reanimated : 2.2.0 => 2.3.1

This induced some conflicts because of required dependencies  of react-native-skeleton-content 1.0.24.

Is it possible to bump versions of these dependencies in a newer version of the package ?
That would be great :)

### LINKED ISSUE #66: Expo SDK 45
**Describe the bug**
This package throws an error when using with Expo SDK 45.

**To Reproduce**
1. Import package in a project using Expo SDK 45
2. Render component
3. See error

**Expected behavior**
This should be compatible with Expo SDK 45.

**Error**
`Error: Requiring module "node_modules/react-native-skeleton-content/lib/index.js", which threw an exception: TypeError: undefined is not an object (evaluating 'NativeUnimoduleProxy.viewManagersNames.includes')`

**Smartphone (please complete the following information):**
 - Device: Iphone 12
 - OS: iOS 14
 - Package version 1.0.24

### PR DESCRIPTION
**Pull request by khevamann**
## Identify the bugs
closes #66 closes #59 closes #33

## Description of the change
I spent a good amount of time updating the packages to newer versions. I started with Reanimated and expo-linear-gradient then removed packages that are deprecated in newer versions of react-native and updated react-native + react.

I also spent some time updating all the dev dependencies to remove npm audit alerts as well as fix the tests failing with newer typescript support

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -21,10 +21,15 @@
     ],
     "@typescript-eslint/no-non-null-assertion": "off",
     "@typescript-eslint/no-explicit-any": "off",
-    "@typescript-eslint/interface-name-prefix": [
+    "@typescript-eslint/naming-convention": [
```
Comment (khevamann) [inline .eslintrc:24]: This rule ws deprecated so I updated to the new recommended approach

File: src/__tests__/SkeletonContent.test.tsx [hunk 2]
```diff
@@ -52,35 +52,37 @@ describe('SkeletonComponent test suite', () => {
     expect(bones[0].props.style).toEqual({
       alignItems: 'center',
       flex: 1,
-      justifyContent: 'center'
+      justifyContent: 'center',
```
Comment (khevamann) [inline src/__tests__/SkeletonContent.test.tsx:55]: If you don't like trailing commas I can remove this, this is just the new default so I left it in

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-skeleton-content",
-  "version": "1.0.24",
+  "version": "1.1.0",
```
Comment (khevamann) [inline package.json:3]: Not sure how you do versioning. Let me know if I should change it to be just "1.0.25"

**Review by khevamann — COMMENTED** [review]
@alexZajac I have been using my own fork of your repo for awhile because it has been broken since Expo 44. I am hoping you will accept this change and bring this package back up to functional with newer expo/react-native versions

File: src/__tests__/SkeletonContent.test.tsx (hunk 2 above)
Comment (alexZajac) [reply src/__tests__/SkeletonContent.test.tsx:55]: No strong opinion, we can keep it!

File: package.json (hunk 3 above)
Comment (alexZajac) [reply package.json:3]: Will probably have to `npm publish` after that, changing the version automatically, we can keep this as it is and will create a follow-up commit to it!

File: src/__tests__/SkeletonContent.test.tsx [hunk 4]
```diff
@@ -352,6 +352,17 @@ describe('SkeletonComponent test suite', () => {
     expect(gradient.props.start).toEqual({ x: 0, y: 0 });
     expect(gradient.props.end).toEqual({ x: 1, y: 0 });
 
+    component.update(
```
Comment (alexZajac) [inline src/__tests__/SkeletonContent.test.tsx:355]: Thanks for making that extra branch!

**Comment by alexZajac** [comment]
Thanks for addressing this, it will unblock a lot more peers!

**Comment by jeff-polls** [comment]
You guys fucking rock 🤘

**Comment by jordann93** [comment]
Hello @alexZajac ,
Do you know when this version will be available / tagged ?

**Pull request merged**